File: src/my_py_pkg/my_py_pkg/number.py

#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class NumberPublisher_node(Node):
    def __init__(self):
        super().__init__('number')
        self.declare_parameter("number_to_publish",5)
        self.declare_parameter('publish_frequency', 11)
        self.publisher_ = self.create_publisher(Int64, 'number', 10)
        self.timer_=self.create_timer(1/self.get_parameter("publish_frequency").value,self.publish_news)
        
    def publish_news(self):
        msg = Int64()
        msg.data = self.get_parameter("number_to_publish").value
        self.publisher_.publish(msg)
        self.get_logger().info('Published number: %d' % msg.data)
       

    
def main(args=None):
    rclpy.init(args=args)

  #  Create instances of NumberPublisher and PubliserSubsriber
    node_num = NumberPublisher_node()
    rclpy.spin(node_num)
    # Clean up
    node_num.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
    

File: src/my_cpp_pkg/src/number.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher() : Node("Number_publisher")
    {
        // Set parameter as int64
        this->declare_parameter("number_to_publish", 777);

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&NumberPublisher::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Number started");
    }

private:
    void publishNews()
    {
        // Get parameter as int64
        int64_t retrieved_param_value;
        this->get_parameter("number_to_publish", retrieved_param_value);

        auto msg = example_interfaces::msg::Int64();
        msg.data = retrieved_param_value;
        publisher_->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}